include/ipr/interface

```diff
@@ -124,19 +124,23 @@ namespace ipr {
    struct Array_delete;          // array delete-expression     delete[] p
    struct Complement;            // bitwise complement                  ~m
    struct Delete;                // delete-expression             delete p
+   struct Demote;                // integral or floating-point conversion
```

Comment:
Let's precise in the comment  "inverse of an integral or floating-point promotion: